aws/data_source_aws_vpc_ids_test.go

```diff
@@ -0,0 +1,90 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+func TestAccDataSourceAwsVpcIDs_basic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceAwsVpcIDsConfig(),
+				Check: resource.ComposeTestCheckFunc(
+					//datasource aws_vpc_ids will list all the vpcs in a region, so it also includes the Default vpc.
```

Comment:
It does not look like anything is reading the `filter` schema attribute into `req.Filter`, which might explain the behavior you're seeing here when you have an invalid filter in the test configuration. It looks like the valid value should be `vpc-id` according to the SDK documentation: https://docs.aws.amazon.com/sdk-for-go/api/service/ec2/#DescribeVpcsInput

For compatibility in all AWS accounts (which may or may not have other existing VPCs since you can delete the default VPC), I would recommend setting up the `_basic` test to not have any ids/filter/tags arguments, create 1 VPC resource, add `depends_on` in the data source to that VPC resource, then ensure that the value of the attribute is greater than 0. We cannot count on `ids` being `"3"`.

Reply:
If you turn on the debug logging, you should see the empty plan output. I generally don't show it during PR submission but I find it very helpful to run my acceptance testing like this:

```
rm -f aws/log.txt; TF_LOG_PATH=log.txt TF_LOG=debug make testacc TEST=./aws TESTARGS='-run=TestAccAWSXXX'
```

Which will turn on debug logging and write that output to a separate log file (`aws/log.txt`) that I can inspect.

Reply:
@bflad Getting the below error when I using `depends_on` in data source to that VPC resource.
`--- FAIL: TestAccDataSourceAwsVpcs_basic (79.22s)
	testing.go:518: Step 0 error: After applying this step and refreshing, the plan was not empty:

		DIFF:

		CREATE: data.aws_vpcs.all
		  ids.#:  "" => "<computed>"
		  tags.%: "" => "<computed>"

		STATE:

		aws_vpc.test-vpc:
		  ID = vpc-954c44ec
		  provider = provider.aws
		  assign_generated_ipv6_cidr_block = false
		  cidr_block = 10.0.0.0/24
		  default_network_acl_id = acl-61b72c19
		  default_route_table_id = rtb-5ff0e927
		  default_security_group_id = sg-29daa658
		  dhcp_options_id = dopt-f0038698
		  enable_classiclink = false
		  enable_classiclink_dns_support = false
		  enable_dns_hostnames = false
		  enable_dns_support = true
		  instance_tenancy = default
		  main_route_table_id = rtb-5ff0e927
		  tags.% = 0
FAIL
FAIL	github.com/terraform-providers/terraform-provider-aws/aws	79.264s`

After destroyed the vpc resource and data source and running the plan shows the drift due to `depends_on` (to vpc resource) in data source. In `data_source_aws_vpc_test.go`, `TestAccDataSourceAwsVpc_basic` fails with same error after adding `depends_on` in data source.  Other tests are fine.